### PR TITLE
Replace PLC4X with etherip

### PR DESCRIPTION
**Pull request by codepitbull**
**Motivation**

Resolves #24713

**Changes**
- Replaces PLC4X with etherip for Ethernet IP/CIP

### CODE REVIEW

File: modules/hivemq-edge-module-etherip/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+repositories {
+    mavenCentral()
+    exclusiveContent {
+        forRepository {
+            maven {
+                url = uri("https://jitpack.io")
+            }
+        }
+        filter {
```
Comment (sfrehse) [inline modules/hivemq-edge-module-etherip/build.gradle.kts:30]: Do we need this within this repo?

File: modules/hivemq-edge-module-etherip/gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+[libraries]
+apache-commonsIO = { module = "commons-io:commons-io", version.ref = "apache-commonsIO" }
+apache-commonsLang = { module = "org.apache.commons:commons-lang3", version.ref = "apache-commonsLang" }
+assertj = { module = "org.assertj:assertj-core", version.ref = "assertj" }
+hivemq-edge-adapterSdk = { module = "com.hivemq:hivemq-edge-adapter-sdk", version.ref = "hivemq-edge-adapterSdk" }
+jackson-databind = { module = "com.fasterxml.jackson.core:jackson-databind", version.ref = "jackson" }
+junit-jupiter = { module = "org.junit.jupiter:junit-jupiter", version.ref = "junit-jupiter" }
+mockito-junitJupiter = { module = "org.mockito:mockito-junit-jupiter", version.ref = "mockito" }
+plc4j-api = { module = "org.apache.plc4x:plc4j-api", version.ref = "apache-plc4x" }
```
Comment (sfrehse) [inline modules/hivemq-edge-module-etherip/gradle/libs.versions.toml:20]: I guess we can clean this up

File: modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    public static final ProtocolAdapterInformation INSTANCE = new EtherIpProtocolAdapterInformation();
+    private static final @NotNull Logger log = LoggerFactory.getLogger(EtherIpProtocolAdapterInformation.class);
+
+    protected EtherIpProtocolAdapterInformation() {
+    }
+
+    @Override
+    public @NotNull String getProtocolName() {
+        return "Ethernet/IP CIP 2";
```
Comment (sfrehse) [inline modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java:44]: I propose we replace the old protocol adapter by the new one and remove the suffx "2" from name and id.

Comment (sfrehse) [reply modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java:44]: This also means we remove the old protocol adapter from the code in order to not have duplicates.

File: modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    @Override
+    public @NotNull String getProtocolId() {
+        return "ethernet-ip-2";
```
Comment (sfrehse) [inline modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java:49]: Needs change

File: modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/model/EtherIpValueFactory.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        try {
+            if(cipData.isNumeric()) {
+                if (cipData.getElementCount() > 1) {
+                    log.warn("More than one element returned, only the first one will be used");
+                }
+                Number number = cipData.getNumber(0);
+                log.debug("Got value {} for type {} for tag address {}", number, dataType, tagAddress);
+                switch (dataType) {
+                    case BOOL:
+                        return Optional.of(new EtherIpBool(tagAddress, number));
```
Comment (sfrehse) [inline modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/model/EtherIpValueFactory.java:30]: Great way of doing it. Making reading every easy.

**Review by sfrehse — COMMENTED** [review]
As I proposed in the comment to remove the old implementation and fix the names such that current configuration just works with the new implementation

File: modules/hivemq-edge-module-etherip/build.gradle.kts (hunk 1 above)
Comment (codepitbull) [reply modules/hivemq-edge-module-etherip/build.gradle.kts:30]: Copy paste mistake.
Fixed.

File: modules/hivemq-edge-module-etherip/gradle/libs.versions.toml (hunk 2 above)
Comment (codepitbull) [reply modules/hivemq-edge-module-etherip/gradle/libs.versions.toml]: Absolutely :)
Fied.

File: modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java (hunk 3 above)
Comment (codepitbull) [reply modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java:44]: Old adapter has been removed.
Names have bee nadjusted.

File: modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java (hunk 4 above)
Comment (codepitbull) [reply modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EtherIpProtocolAdapterInformation.java]: Fixed.

File: modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/model/EtherIpValueFactory.java (hunk 5 above)
Comment (codepitbull) [reply modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/model/EtherIpValueFactory.java:30]: Thanks :)

**Review by sfrehse — APPROVED** [review]

**Pull request merged**